### PR TITLE
MINOR: Remove Scala Procedure Syntax

### PR DESCRIPTION
**Pull request by jolshan**
Scala 2.13 deprecates procedure syntax. This means that for every usage, there will be a build warning stating this. There are 1800+ instances of procedure syntax in Kafka, and this PR attempts to remove them.

### CODE REVIEW

File: core/src/main/scala/kafka/utils/KafkaScheduler.scala [hunk 1]
```diff
@@ -32,14 +32,12 @@ trait Scheduler {
...
    */
-  def startup()
-  
+  def startup(): Unit
   /**
    * Shutdown this scheduler. When this method is complete no more executions of background tasks will occur. 
    * This includes tasks scheduled with a delayed execution.
    */
-  def shutdown()
-
```
Comment (stanislavkozlovski) [inline core/src/main/scala/kafka/utils/KafkaScheduler.scala]: Is there a reason for the space removal here? I would think it reads better when there is some space in between

File: core/src/test/scala/integration/kafka/api/BaseQuotaTest.scala [hunk 2]
```diff
@@ -168,9 +168,8 @@ abstract class QuotaTestClients(topic: String,
                                 val consumer: KafkaConsumer[Array[Byte], Array[Byte]]) {
 
   def userPrincipal: KafkaPrincipal
-  def overrideQuotas(producerQuota: Long, consumerQuota: Long, requestQuota: Double)
-  def removeQuotaOverrides()
-
```
Comment (stanislavkozlovski) [inline core/src/test/scala/integration/kafka/api/BaseQuotaTest.scala]: nit: ditto for the space. I would personally prefer if we put spaces in between the methods

**Review by stanislavkozlovski — COMMENTED** [review]
Thanks for the PR! I really appreciate how you also removed the extra spaces in instances like `method() : Unit = {}`

I see 3 more warnings in `ServerGenerateClusterIdTest.scala`. Let's fix them too. Please don't force-push anymore since I've reviewed the lengthy commits and would like a small diff to look at next time.

I left some small nits about spaces in between methods. I saw in other code that we seem to not have any space between abstract methods, so I assume I may be going against the convention

File: core/src/main/scala/kafka/utils/KafkaScheduler.scala (hunk 1 above)
Comment (jolshan) [reply core/src/main/scala/kafka/utils/KafkaScheduler.scala]: That was a minor mistake on my part. Fixing.

**Comment by jolshan** [comment]
Thanks for the speedy review! I've made the suggested changes.

**Review by stanislavkozlovski — APPROVED** [review]
LGTM! Thanks

**Comment by cmccabe** [comment]
LGTM.  Thanks, @jolshan

**Pull request merged**